### ISSUE TITLE
If datasource source is specified no fields are included by default

### ISSUE THREAD

**Issue by izderadicka** [nicolaiarocci/eve#842]
Documentation says:
projection 	Fieldset exposed by the endpoint. If omitted, by default all fields will be returned to the client. 

However if projection is missing from datasource quite opposite happens - no fields are included (apart _id, _etag ....)

**Comment by nicolaiarocci** [comment]
Hello, on which Eve version are you?

**Comment by izderadicka** [comment]
'0.6.3'  on python 3.4

I.

**Comment by nicolaiarocci** [comment]
Can you please post your `datasource` setting? I can't reproduce this.

**Comment by izderadicka** [comment]
Still have this issue in develop branch:

```python
series={
        'datasource': {
            'source':'ebooks',
            }
        }
        
DOMAIN={'ebooks': ebooks,
        'authors': authors,
        'series': series}
```

And this is what I get in browser for http://localhost:5000/series:
```
<resource href="series" title="series">
<link rel="last" href="series?page=1317" title="last page"/>
<link rel="next" href="series?page=2" title="next page"/>
<link rel="parent" href="/" title="home"/>
<_meta><max_results>25</max_results><page>1</page><total>32910</total></_meta>
<resource href="series/56f24523e67883403c47833f" title="Serie">
<_created>Thu, 01 Jan 1970 00:00:00 GMT</_created>
<_etag>ca0773ec15ea1629194b73d4168ad96762f6a42d</_etag><_id>56f24523e67883403c47833f</_id>
<_updated>Thu, 01 Jan 1970 00:00:00 GMT</_updated>
</resource>
<resource href="series/56f24523e67883403c478340" title="Serie">
<_created>Thu, 01 Jan 1970 00:00:00 GMT</_created>
<_etag>64ab7241e966331b34604a5184438932f4817916</_etag>
<_id>56f24523e67883403c478340</_id>
<_updated>Thu, 01 Jan 1970 00:00:00 GMT</_updated>
</resource>
...
```

**Comment by nicolaiarocci** [comment]
You don't have any callback function hooked to your GET method, do you? It might be altering the response payload.

**Comment by nicolaiarocci** [comment]
Also, what is the `schema` for the series endpoint?

**Comment by izderadicka** [comment]
No callback function, nothing fancy.   The only thing that is special is that documents were loaded via mongoimport not created via API.

There is no schema for that particular endpoint -   just uses ebooks collection. Sample documents attached.
[test-ebooks.json.zip](https://github.com/nicolaiarocci/eve/files/189223/test-ebooks.json.zip)


For ebooks this schema is used - but is it not probably relevan:
```python
LANGUAGES={'cs':'Czech',
           'en': 'English',
           'ru': 'Russian',
           'sk': 'Slovak',
           }
FORMATS={
    'pdf':('application/pdf', 'PDF document'),
    'txt':('text/plain', 'Plain text document'),
    'pdb':('application/vnd.palm', 'Palm Document'),
    'rtf':('application/rtf', 'RTF Document'),
    'doc':('application/msword', 'MS Word Document'),
    'mobi':('application/x-mobipocket-ebook', 'Mobipocket Ebook'),
    'prc':('application/vnd.palm ', 'Palm/Mobipocket Document'),
    'epub':('application/epub+zip', 'Open Publication Structure eBook'),
    'odt':('application/vnd.oasis.opendocument.text ', 'OpenOffice Document'),
    'djvu':('image/vnd.djvu', 'DJVU Document'),
    'html':('text/html', 'HTML File'),
    'htm':('text/html', 'HTML'),
    'chm':('application/octet-stream', 'Compiled HTML'),
    'lit':('application/x-ms-reader ', 'MS Reader Ebook'),
    'fb2':('text/xml', 'Fiction Book 2'),
    'docx':('application/vnd.openxmlformats-officedocument.wordprocessingml.document', 'MS Word XML document'),
    'azw3':('application/x-mobi8-ebook', 'Amazon KF8 eBook File'),
    }

source={'type':'dict',
        'schema':{
            '_id': {
                'type':'objectid',
                'required':True
                },
            'format': {
                'type':'string',
                'minlength':1,
                'maxlength':20,
                'allowed':list(FORMATS.keys()),
                'required':True,
                },
            'size': {
                'type': 'integer',
                'required':True,
                },
            'hash': {
                'type':'string',
                'maxlength':64,
                'minlength':4,
                'required':True
                  },
            'location': {
                'type':'string',
                'maxlength':1024,
                'minlength':4,
                'required': True
                },
            'original_name': {
                'type':'string',
                'maxlength':1024,
                'minlength':4,
                'required': True
                },
            'quality': {
                'type':'number',
                'min':0,
                'max':100,
                'nullable':True
                }
            }
        }
author={'type': 'dict',
       'schema': {
            'lastname': {
                'type':'string',
                'minlength':1,
                'maxlength':255,
                'required':True 
                },
            'firstname': {
                'type':'string',
                'maxlength': 255,
                'required':False,
                },
            }
        }

ebooks={
        'item_title':'ebook',
        'schema': {
                   'title': {
                       'type':'string',
                       'minlength':1,
                       'maxlength':255,
                       'required':True,
                       },
                   'authors': {
                        'type':'list',
                        'schema': author,
                        'maxlength':100
                    },
                    'series': {
                        'type':'string',
                        'maxlength':255,
                        'nullable': True
                        },
                    'series_index':{
                        'type':'integer',
                        'max':999,
                        'nullable': True
                        },
                    'rating': {
                        'type':'number',
                        'min':0,
                        'max':100,
                        'nullable': True,
                        },
                    'language': {
                        'type':'string',
                        'minlength':2,
                        'maxlength':2,
                        'allowed': list(LANGUAGES.keys()),
                        'required':True
                        },
                    'genres' : {
                        'type':'list',
                        'maxlength':20,
                        'schema': {
                            'type':'string',
                            'minlength':1,
                            'maxlength':64,
                            },
                        },
                    'sources': {
                        'type':'list',
                        'schema': source,
                        'maxlength':100
                        }
            }
        }

```

**Comment by nicolaiarocci** [comment]
> There is no schema for that particular endpoint

Ok, thanks. That's what triggers it.

**Comment by allanberry** [comment]
Did this fix make it in by v0.7?  I'm still seeing this behavior.

(I'm a n00b with Eve, though, so maybe I'm doing something else incorrectly.)

**Comment by allanberry** [comment]
@nicolaiarocci  I made an [SO question](http://stackoverflow.com/q/42382645/652626) about my issue; possible user error, but I figure I should mention in case this bug has inadvertently cropped up again.  If you're busy, I can look into the source for a PO, but I'm kind of a rookie with that sort of thing: perhaps you can give me a hint where to begin.

By the way, thanks regardless; Eve is beautiful software.  I'm really enjoying it so far.
